Assets/Scripts/EnemySpawnManager.cs

```diff
@@ -0,0 +1,40 @@
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+public class EnemySpawnManager : MonoBehaviour
+{
+    // Game object that the spawner will be attached to
+    [SerializeField] GameObject attachedGameObject;
+    [SerializeField] float distanceToAttachedObject;
+    [SerializeField] private GameObject enemyPrefab;
+    [SerializeField] private float startDelay;
+    [SerializeField] private float repeatedRate;
+    [SerializeField] private float minYPos;
+    [SerializeField] private float maxYPos;
+
+    private void Start()
+    {
+        InvokeRepeating("SpawnEnemy", startDelay, repeatedRate);
+    }
+
+    void Update()
+    {
+        AdjustSpawnerPosition();
+    }
+
+    void SpawnEnemy()
+    {
+        Transform gameObjectPos = gameObject.transform;
+        float randomYPos = Random.Range(minYPos, maxYPos);
+        Instantiate(enemyPrefab, new Vector3(gameObjectPos.position.x, randomYPos, gameObjectPos.position.z), Quaternion.identity);
+    }
+
+    // Adjusts the spawner to a given distance in front of the given game object
+    void AdjustSpawnerPosition()
```

Comment:
This is adjusting the spawn point stay a certain distance from the given object.

Reply:
Oh yeah, but why is it necessary, is what I meant. It seems like you're having enemies spawn an x amount of distance away from the Player game object right? I think it would be easier to have the enemies spawn offscreen at a defined x position